modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserLocalServiceTest.java

```diff
@@ -91,6 +97,44 @@ public class UserLocalServiceTest {
 	public static final AggregateTestRule aggregateTestRule =
 		new LiferayIntegrationTestRule();
 
+	@Test
+	public void testAuthenticateByEmailAddressWithPasswordPolicy()
+		throws Exception {
+
+		String password = "password";
+		String newPassword = RandomTestUtil.randomString();
+
+		User user = _addRandomUserWithPassword(password);
+
+		Assert.assertEquals(
+			Authenticator.FAILURE,
+			_userLocalService.authenticateByEmailAddress(
+				user.getCompanyId(), user.getEmailAddress(), newPassword, null,
+				null, null));
+
+		int failedLoginAttempts = user.getFailedLoginAttempts();
+
+		user = _userLocalService.updatePassword(
+			user.getUserId(), newPassword, newPassword, false, true);
+
+		PasswordPolicy passwordPolicy = user.getPasswordPolicy();
+
+		passwordPolicy.setExpireable(true);
+		passwordPolicy.setMaxAge(0);
+
+		PasswordPolicyLocalServiceUtil.updatePasswordPolicy(passwordPolicy);
+
+		try {
+			_userLocalService.authenticateByEmailAddress(
+				user.getCompanyId(), user.getEmailAddress(), newPassword, null,
+				null, null);
+		}
+		catch (PasswordExpiredException passwordExpiredException) {
+			Assert.assertEquals(
+				failedLoginAttempts + 1, user.getFailedLoginAttempts());
+		}
```

Comment:
I think it should be:
```
	catch (PortalException portalException) {
		Assert.assertEquals(
			PasswordExpiredException.class, portalException.getClass());
	}
```
and should move 
```
	Assert.assertEquals(
		failedLoginAttempts + 1, user.getFailedLoginAttempts());
```
out of the catch block and should fetch the user again before Assert.assertEquals.